streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -338,6 +369,41 @@ abstract class DStream[T: ClassTag] (
   }
 
   /**
+   * Wrap a body of code such that the call site and operation scope
+   * information are passed to the RDDs created in this body properly.
+   */
+  protected def createRDDWith[U](time: Time)(body: => U): U = {
```

Comment:
the name is incomplete. `createRDDWith` does not make sense. It has to `withSomething`. Maybe `withEnv`?
Also why did you change it from `doCompute`? Is there are point in wrapping whole of `getOrCompute` in this function, rather than just the `compute()` function as you were doing it earlier with `doCompute`?